pode/pode.py

```diff
@@ -1,1464 +1,785 @@
-    if edge is not None:
-        polygon = order_by_edge(polygon, edge=edge)
-    else:
-        polygon = order_by_site(polygon, sites=sites)
-    graph = nx.relabel_nodes(graph, {original_polygon: polygon})
-    if with_gradual_search and len(sites) == 1:
-        partitions_to_process = gradual_search(polygon=polygon,
-                                               sites=sites,
-                                               edge=edge)
-    else:
-        partitions_to_process, graph, neighbor = nonconvex_divide(polygon,
-                                                                  sites,
-                                                                  graph=graph)
-    partitions_union = unary_union([partition.polygon
-                                    for partition in partitions_to_process])
-    nodes_to_remove = [node for node in graph
-                       if partitions_union.contains(node.buffer(-1e-9))]
-    graph.remove_nodes_from(nodes_to_remove)
-    if len(partitions_to_process) == 3:  # Case 1.3
-        return prepend_partitions_inline(partitions_to_process,
-                                         graph=graph,
-                                         node_to_attach_to=neighbor)
-    else:  # Cases 1.1 and 1.2
-        return prepend_partitions(partitions_to_process,
-                                  graph=graph,
-                                  node_to_attach_to=neighbor)
+    ordered_points = sorted(points.points, key=angle)
+    site_index = next(index for index in range(len(ordered_points))
+                      if ordered_points[index] == site_location)
+    return ordered_points[site_index + 1:] + ordered_points[:site_index + 1]
 
 
-def add_sites(polygon: Polygon,
-              sites: SitesType) -> Polygon:
+def order_by_edge(vertices: Multipoint,
+                  edge: Segment) -> List[Point]:
     """
-    Adds sites to polygon vertices for further iteration over them
-    in the nonconvex division part of the algorithm.
+    Orders vertices of the convex polygon and the sites in a
+    counterclockwise manner so that the last edge would be the edge to
+    the next neighbor.
+    :param vertices: vertices of the convex polygon and the sites
+    located on the boundary of the polygon
+    :param edge: edge to the next neighbor
+    :return: ordered union of polygon vertices and sites
     """
-    polygon_ring = polygon.exterior
-    distance_along_vertices: Callable[[Point], float] = polygon_ring.project
-    vertices = MultiPoint(polygon_ring.coords[:-1])
-    sites_points = MultiPoint([*sites])
-    vertices_and_sites = vertices.union(sites_points)
-    sorted_points = sorted(vertices_and_sites.geoms,
-                           key=distance_along_vertices)
-    # workaround for a bug https://github.com/Toblerity/Shapely/issues/706
-    return Polygon(LineString(sorted_points))
+    points = cast(Multipoint, vertices | Multipoint(edge.start, edge.end))
+    centroid_ = centroid(points)

```

Comment:
it looks like with the latest version of `gon` we can take rid of `centroid` function